src/fstab-generator/fstab-generator.c

```diff
@@ -714,6 +714,12 @@ static int add_sysroot_mount(void) {
                 return 0;
         }
 
+        if (startswith(arg_root_what, "nfs:")) {
```

Comment:
The dracut docs say:

       root=[<server-ip>:]<root-dir>[:<nfs-options>]
           mount nfs share from <server-ip>:/<root-dir>, if no server-ip
           is given, use dhcp next_server. If server-ip is an IPv6
           address it has to be put in brackets, e.g. [2001:DB8::1]. NFS
           options can be appended with the prefix ":" or "," and are
           separated by ",".

       root=nfs:[<server-ip>:]<root-dir>[:<nfs-options>],
       root=nfs4:[<server-ip>:]<root-dir>[:<nfs-options>],
       root={dhcp|dhcp6}

I think we'd want to support all of those. Checking for "nsf:", "nsf4:", "dhcp", "dhcp6" should be easy.
The case where `server-ip` is given with no 'nfs/nfs4' prefix could be detected by trying to parse the IP.
Those docs say that specifying just the `root-dir` alone is also allowed, but I think that this case is impossible to ignore for a normal device path, so we should just ignore it.

Reply:
> Those docs say that specifying just the `root-dir` alone is also allowed, but I think that this case is impossible to ignore for a normal device path, so we should just ignore it.

Is it possible to specify non device path without requesting nfsroot, e.g. /foo/bar? I guess we can assume that non device path is NFS root.